doc/howtos/rdtraining/compute_onchange.rst

```diff
@@ -0,0 +1,291 @@
+.. _howto/rdtraining/compute_onchange:
+
+=============================
+Computed Fields And Onchanges
+=============================
+
+The :ref:`relations between models <howto/rdtraining/relations>` is a key component of any Odoo
+module. It is necessary to the modelization of any business case: models are linked one to another.
+However, we might want links between the fields inside of a given model. Sometimes because the
+value of a field is determined from the value of other fields, sometimes to help the user encoding.
+
+Although this chapter is not technically complex, the semantic of both concepts is very important.
+This is also the first time we will write Python logic. Up to now, we never wrote anything else
+that class definiton or field declaration.
+
+Computed Fields
+===============
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/fields/compute`.
+
+.. note::
+
+    **Goal**: at the end of this section:
+
+    - On the property model, the total area and the best offer are computed:
+
+    .. image:: compute_onchange/media/compute.gif
+        :align: center
+        :alt: Compute fields
+
+    - On the property offer model, the validity date is computed and has an inverse:
+
+    .. image:: compute_onchange/media/compute_inverse.gif
+        :align: center
+        :alt: Compute field with inverse
+
+In our real estate module, we have defined the living area as well as the garden area. It is then
+natural to define the total area as the sum of both fields. We will use the concept of computed
+field for this: the value of a given field is computed from the value of other fields.
+
+So far fields have been stored directly in and retrieved directly from the
+database. Fields can also be *computed*. In that case, the field's value is not
+retrieved from the database but computed on-the-fly by calling a method of the
+model.
+
+To create a computed field, create a field and set its attribute
+:attr:`~odoo.fields.Field.compute` to the name of a method. The computation
+method should simply set the value of the field to compute on every record in
```

Comment:
We can precise that the convention for the method is to start with '_compute' (as it's done in the previous chapter for the fields finishing with _id or _ids)